### DISCUSSION TITLE
Price of an NFT purchased by a token other than ALGO

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#334] [General]
The focus in our development lately has been on improving our NFT engine and updating missing NFT markets and their functionalities.

The process is straightforward and after it is "completed" (no completion in praxis, the ecosystem is in an everlasting changing process)  it will lead to filling enough data for our NFT price engine driven by AI.

One of the topics for the development requires a public discussion, and it's related to purchases and listings of NFTs using some other currency. One of the obvious examples is [FIFA+ Collect](https://collect.fifa.com/) and their USDC/US$ only marketplace, but some others accept payments in USDC like [Abris](https://abris.io/) for example. Also, there are some NFT markets like [Algogems](https://www.algogems.io/) and brand new [NFT Factory](https://nftfactory.org/) which accept a variety of tokens for payments.

That being said, what should be our policy for keeping such price records? Should we collect and keep them in the original currency and present them in ALGO (or USD) at the time when they are revealed on an address page or we should record them in ALGO from the very start?

If you bought your FIFA+ Collect NFT for US$1.67 at the time when 1 ALGO was worth US$0.30, do you expect that we show 5.55 ALGO under "Last purchase" or the correct way is to show 4.27 ALGO by counting the current price for ALGO of US$0.39?

**Comment by marucat27** [comment]
For the FIFA NFTs, I’m thinking to show the price at the time of sale but in current value in ALGO.  Since it will show the value in USD the person paid for the NFT.  I’m thinking …

**Reply by ipaleka** [reply]
> For the FIFA NFTs, I’m thinking to show [...]

FIFA+ Collect is presented just as an example NFT market, we simply have to define the same rule that will apply to every NFT market and every NFT.

**Reply by ipaleka** [reply]
> I’m thinking to show the price at the time of sale but in current value in ALGO. Since it will show the value in USD the person paid for the NFT. I’m thinking …

Yes, that goes in line with the whole ecosystem policy and our engine in particular: at some point in time, you bought some ASA (like Smile, YLDY, ...) and we simply don't use the price at that particular time/moment for any other purpose except maybe for bragging or complaining, respectively to calculate our gains and losses. Its value is always presented as of now.
